sdk/src/userMap/userMap.ts

```diff
@@ -303,65 +303,59 @@ export class UserMap implements UserMapInterface {
 				},
 			];
 
-			const rpcJSONResponse: any =
-				// @ts-ignore
-				await this.connection._rpcRequest('getProgramAccounts', rpcRequestArgs);
-
-			const rpcResponseAndContext: RpcResponseAndContext<
-				Array<{
-					pubkey: PublicKey;
-					account: {
-						data: [string, string];
-					};
-				}>
-			> = rpcJSONResponse.result;
-
+			// @ts-ignore
+			const rpcJSONResponse: any = await this.connection._rpcRequest('getProgramAccounts', rpcRequestArgs);
+			const rpcResponseAndContext: RpcResponseAndContext<Array<{ pubkey: PublicKey; account: { data: [string, string]; } }>> = rpcJSONResponse.result;
 			const slot = rpcResponseAndContext.context.slot;
-
 			this.updateLatestSlot(slot);
 
 			const programAccountBufferMap = new Map<string, Buffer>();
-			for (const programAccount of rpcResponseAndContext.value) {
+			rpcResponseAndContext.value.forEach(programAccount => {
```

Comment:
Have you benchmarked this change against whatever the average response length received is?

Last I checked `forEach` was the least performant way to iterate in Javascript.

Reply:
i dont think it makes much difference with for...of, main improvement in term of execution time i have benchmarked when calling userMap.subscribe using keeper bot went from 120 seconds to 11 on my local machine was due to parallel processing, which is right after this iteration. i apologize in advance im still learing and improving